### PR TITLE
render in dock

### PR DESCRIPTION
**Pull request by kaleidoc0de**
This pr takes the work done [here](https://github.com/platformio/platformio-atom-ide-terminal/pull/697)
with a few refactors
adds the option to render terminals in the bottom dock

### CODE REVIEW

**Comment by the-j0k3r** [comment]
@kaleidoc0de thanks you for this PR original code by @akonwi

There are still issues related to this Windows side documented at https://github.com/platformio/platformio-atom-ide-terminal/pull/697#issuecomment-514521563

I would need to retest this to see if that issue is resolved.

@kaleidoc0de 

#### Please note the work currently being done at https://github.com/bus-stop/terminus/tree/decaffeinated-terminus

This is a JavaScript version of the project, that will become the new master branch of this project pending result of a few remaining tests.

If that becomes true this means this PR will need to be submitted in JavaScript. Ive already notified @akonwi at slack of this.

Please bare with us while we move to JavaScript, I hope this is good news too.

File: lib/view.coffee [hunk 1]
```diff
@@ -250,7 +280,20 @@ class TerminusView extends View
...
+  openInDock: =>
+    @statusBar.setActiveTerminalView this
+    @statusIcon.activate()
+    atom.workspace.open(this, activatePane: true)
+    .then =>
+      if not @opened
+        @opened = true
+        @displayTerminal()
+        @xterm.height("100%")
+        @emit "terminusl:terminal-open"
```
Comment (kaleidoc0de) [inline lib/view.coffee]: typo

Comment (the-j0k3r) [inline lib/view.coffee]:
```suggestion
        @emit "terminus:terminal-open"
```

Comment (the-j0k3r) [reply lib/view.coffee]: Please check this box on sidebar

Capture

Comment (the-j0k3r) [reply lib/view.coffee]: Please check this box on the pull request sidebar

Capture

**Comment by the-j0k3r** [comment]
@kaleidoc0de We're changing master branch to the JavaScript branch, so renaming branches closed pull request unexpectedly.

You can resubmit this PR to master-old branch if you wish but with move to Javascript this is better in long run.

**Comment by the-j0k3r** [comment]
Changeover is completed, thank you for your patience.

**Pull request closed without merging**